Library/Homebrew/cmd/audit.rb

```diff
@@ -747,6 +748,20 @@ def audit_caveats
     end
   end
 
+  def audit_reverse_migration
+    # Only enforce for new formula being re-added to core
+    return unless @strict
+    return unless formula.tap.downcase == "homebrew/homebrew"
+
+    unless TAP_MIGRATIONS.map(&:to_s).grep(/#{formula.name}/).empty?
```

Comment:
I think it is best to avoid such string matching.  It's bad for performance and likeky to break if there's any change in upper layer data structure.